src/ui/public/visualize/visualize.js

```diff
@@ -53,7 +53,12 @@ uiModules
         const responseHandler = getHandler(responseHandlers, $scope.vis.type.responseHandler);
 
         $scope.fetch = _.debounce(function () {
-
+          if (!$scope.vis.initialized) {
```

Comment:
we need to wait for vis to be initialized before running the request handler

Reply:
What about using `$scope.$watch('vis.initialized', $scope.fetch);` instead, so we don't have to poll? Then we can get rid of the destroy timeout extra code too.